src/converse-register.js

```diff
@@ -349,6 +371,9 @@
                         $form.append('<input type="button" class="submit" value="'+__('Return')+'"/>');
                         $form.find('input[type=button]').on('click', this.cancelRegistration.bind(this));
                     }
+                    if (_converse.registration_domain !== '') {
+                        $form.find('input[type=button]').hide();
+                    }
```

Comment:
No this cannot be done from the template. There is no template seperate template file for this. If you see the code for the function above this line, the button is rendered in the function itself, by passing the entire html string. So this is the best method to hide it here.